### PR TITLE
[Uid] Use CPP

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | --
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Uid/Command/GenerateUlidCommand.php [hunk 1]
```diff
@@ -25,12 +25,8 @@
 #[AsCommand(name: 'ulid:generate', description: 'Generate a ULID')]
 class GenerateUlidCommand extends Command
 {
-    private UlidFactory $factory;
-
-    public function __construct(UlidFactory $factory = null)
+    public function __construct(private UlidFactory $factory = new UlidFactory())
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Uid/Command/GenerateUlidCommand.php]: let's always put CPP properties on their own line, that'll reduce also the diff when adding new arguments in the future

Comment (OskarStark) [reply src/Symfony/Component/Uid/Command/GenerateUlidCommand.php]: I like it 👍🏻

Comment (OskarStark) [reply src/Symfony/Component/Uid/Command/GenerateUlidCommand.php]: Done

File: src/Symfony/Component/Uid/Command/GenerateUlidCommand.php [hunk 2]
```diff
@@ -25,12 +25,9 @@
...
 class GenerateUlidCommand extends Command
 {
-    private UlidFactory $factory;
-
-    public function __construct(UlidFactory $factory = null)
-    {
-        $this->factory = $factory ?? new UlidFactory();
-
+    public function __construct(
+        private UlidFactory $factory = new UlidFactory(),
```
Comment (fancyweb) [inline src/Symfony/Component/Uid/Command/GenerateUlidCommand.php:29]:
```suggestion
        private readonly UlidFactory $factory = new UlidFactory(),
```
Can we also add readonly where we can?

Comment (OskarStark) [reply src/Symfony/Component/Uid/Command/GenerateUlidCommand.php:29]: I think @nicolas-grekas is 👎 on this

Comment (nicolas-grekas) [reply src/Symfony/Component/Uid/Command/GenerateUlidCommand.php:29]: It'd be useless or just visual debt in practice to me yes.

**Review by fancyweb — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @OskarStark.

**Pull request merged**